### DISCUSSION TITLE
Prisma postgresql migrations error on `column "mycolumn" contains null values` with exiting data

### DISCUSSION THREAD

**Discussion by sandun86** [prisma/prisma#18649] [Q&A]
I am using `Prisma` PostgreSQL migrations for my project. So I am managing a `schema.prisma` file.

1. Firstly I have created the following table

```
model WaitlistRecords {
  id         Int      @id @default(autoincrement())
  first_name String   @db.VarChar(100)
  last_name  String   @db.VarChar(100)
  mobile     String
  email      String
  status     String
  priority   String
}
```
2. Inserted the data into the table
3. Now I need to rename the `priority` column as `cohorts`. 
4. following is the new table with changes 
```
model WaitlistRecords {
  id         Int      @id @default(autoincrement())
  first_name String   @db.VarChar(100)
  last_name  String   @db.VarChar(100)
  mobile     String
  email      String
  status     String
  cohorts   String
}
```
5. so I have created the migration file using this command `npx prisma migrate dev --name renameCityColumnRecordsTable`
6. migration is running successfully in the `empty table`
7. when **I am running with the existing data it is getting the following error ---->. So how can I solve this problem? Please help me. 
Thanks 
![image](https://user-images.githubusercontent.com/4927289/230366062-63ed615e-1c62-4d67-a9da-92908ff667e3.png)

**Comment by nurul3101** [comment]
Hello @sandun86 👋 

I can see that there is no column named "mycolumn" in the records model. Do you get this exact error message?

> column "mycolumn" contains null values

or "mycolumn" is just a place holder and you get an error message with different column name.

**Reply by sandun86** [reply]
Thanks for your reply. please revisit the issue. I have updated it

**Reply by sandun86** [reply]
And this is the generated migration query 
```
ALTER TABLE "WaitlistRecords" DROP COLUMN "priority",
ADD COLUMN     "cohorts" TEXT NOT NULL;
```

**Reply by nurul3101** [reply]
Apologies for the delay in getting back to you. 

I believe the error is occurring because you're renaming the priority column to cohorts, and the existing data in the table has some null values in the priority column. Since the new cohorts column does not have a default value specified, the database does not know how to handle the null values. 

You would need to manually fix those null values before running the migration.

**Reply by nurul3101** [reply]
If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by sandun86** [reply]
Apologies for the delayed reply.
Actually `priority column` does not have `null` values.  I have already checked that scenario.

**Reply by sandun86** [reply]
Hi @nurul3101 I have found a solution on the doc :). I have missed it 
https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/customizing-migrations#example-rename-a-field
Anyway, thanks for your support. 👍👍
